src/app/api/services/push-publish/push-publish.service.ts

```diff
@@ -0,0 +1,96 @@
+import { CoreWebService, ApiRoot } from 'dotcms-js/dotcms-js';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { RequestMethod } from '@angular/http';
+
+/**
+ * Provide method to push publish to content types
+ * @export
+ * @class PushPublishService
+ */
+@Injectable()
+export class PushPublishService {
+    private pushEnvironementsUrl: string;
+    private currentUsersUrl: string;
+    private publishUrl: string;
+
+    constructor(public _apiRoot: ApiRoot, private coreWebService: CoreWebService) {
+        this.pushEnvironementsUrl = `${this._apiRoot.baseUrl}api/environment/loadenvironments/roleId`;
+        this.currentUsersUrl =  `${this._apiRoot.baseUrl}api/v1/users/current/`;
+        this.publishUrl = `${this._apiRoot.baseUrl}DotAjaxDirector/com.dotcms.publisher.ajax.RemotePublishAjaxAction/cmd/publish`;
+    }
+
+    /**
+     * Get environments and return observable of environments filtered by not empty name
+     * @returns {Observable<any>}
+     * @memberof PushPublishService
+     */
+    getEnvironments(): Observable<any> {
+        return this.getCurrentUser().flatMap(user => {
+            return this.coreWebService.request({
+                method: RequestMethod.Get,
+                url: `${this.pushEnvironementsUrl}/${user.roleId}/name=0`
+            });
+        })
+        .map(environments => environments.filter(environment => environment.name !== ''));
+    }
+
+    /**
+     * Do a Post request with push publish data and returns an object with errormessage, total, bundleId and errors
+     * @param {string} contentTypeId
+     * @param {*} formValue
+     * @returns {Observable<PushPublishResponse>}
+     * @memberof PushPublishService
+     */
+    pushPublishContent(contentTypeId: string, formValue: any): Observable<PushPublishResponse> {
+        return this.coreWebService.request({
+            body: this.getPublishEnvironmentData(contentTypeId, formValue),
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded'
+            },
+            method: RequestMethod.Post,
+            url: this.publishUrl
+        });
+    }
+
+    /**
+     * Do a Get request to get current users and returns an observable with current users
+     * @returns {Observable<any>}
+     * @memberof PushPublishService
+     */
+    getCurrentUser(): Observable<any> {
```

Comment:
Because we don't have a "user service" we have the login service but is in dotcms-js and the fix it's to make the User return the role id, so for now I told him to put it here.